lmfdb/static/lmfdb.js

```diff
@@ -518,40 +518,6 @@ function resetStart()
   // this will be cleaned by the cleanSubmit
 }
 
-function cleanSubmit(id)
-{
-  var myForm = document.getElementById(id);
-  var allInputs = myForm.getElementsByTagName('input');
-  var allSelects = myForm.getElementsByTagName('select');
-  var item, i, n = 0;
-  for(i = 0; item = allInputs[i]; i++) {
-    if (item.getAttribute('name') ) {
-        // Special case count so that we strip the default value
-        if (!item.value || (item.getAttribute('name') == 'count' && item.value == 50)) {
-        item.setAttribute('name', '');
-      } else {
-        n++;
-      }
-    }
-  }
-  for(i = 0; item = allSelects[i]; i++) {
-    if (item.getAttribute('name') ) {
-      if (!item.value) {
-        item.setAttribute('name', '');
-      } else {
-        n++;
-      }
-    }
-  }
-  if (!n) {
-    var all = document.createElement('input');
-    all.type='hidden';
-    all.name='all';
-    all.value='1';
-    myForm.appendChild(all);
```

Comment:
which pages is this relevant for now?

Reply:
I think that was just there in order to make things work properly when the user didn't enter any selections (it is only set in that case).  I think you should be able to remove it now (the thing to test would simply be that searches with no selections work).

Reply:
Yeah i tried a couple and it seemed that the search type = List was always set so that seems ok. I'll try a search on every subdirectory of the lmfdb later